archunit/src/test/java/com/tngtech/archunit/core/domain/InstanceofCheckTest.java

```diff
@@ -0,0 +1,95 @@
+package com.tngtech.archunit.core.domain;
+
+import java.util.Iterator;
+
+import org.assertj.core.api.iterable.Extractor;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static com.tngtech.archunit.core.domain.TestUtils.importClasses;
+import static com.tngtech.archunit.core.domain.TestUtils.importMethod;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class InstanceofCheckTest {
+
+    @Test
+    public void method_contains_instanceof_type() {
+        JavaMethod method = importMethod(SomeClass.class, "method", Object.class);
+
+        assertThat(method.getInstanceofChecks()).hasSize(1);
+        Iterator<InstanceofCheck> iterator = method.getInstanceofChecks().iterator();
+        InstanceofCheck instanceofCheck = iterator.next();
+
+        assertThat(instanceofCheck.getRawType().getSimpleName()).as("instanceof check target is " + C.class.getSimpleName()).isEqualTo(C.class.getSimpleName());
+    }
+
+    @Test
+    public void class_contains_all_instanceof_dependencies() {
+        JavaClass javaClass = importClasses(SomeClass.class).get(SomeClass.class);
+
+        assertThat(javaClass.getDirectDependenciesFromSelf()).extracting(simpleNameOfTargetClass()).contains(A.class.getSimpleName(), B.class.getSimpleName(), C.class.getSimpleName());
+    }
+
+    @Test
+    @Ignore("Static initializers are not registered in MemberDependenciesByTarget")
```

Comment:
The reason is, that it either made no sense (e.g. when parameters, return types, throws declarations, etc., are added, none of those are existing for a static initializer), or there is not enough info at the time (accesses of various type, at the point of registering those, we only have some raw string `desc` but the real "target" `JavaMember` is constructed at a way later point in time; you can see how that is handled within `JavaClassProcessor.AccessHandler` and `DomainObjectCreationContext.AccessContext`)
`InstanceofChecks` are just different in semantics. While the other ones are part of the declaration (parameters, return type, ...) `InstanceofChecks` are statements and reside within the bode of a `JavaCodeUnit`.
What do you think about adding a method `ClassGraphCreator.MemberDependenciesByTarget.registerStaticInitializer(..)` and registering only the `InstanceofChecks` by target there?
You are definitely right that this test should pass, so good catch!! :+1: